File: accounts/accounts-api/src/main/java/com/hrrm/famoney/accounts/api/EntryItemDataDTO.java
package com.hrrm.famoney.accounts.api;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.hrrm.famoney.commons.immutables.ImmutableStyle;
import io.swagger.v3.oas.annotations.media.Schema;
import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.math.BigDecimal;

@Schema(name = "EntryItemData")
@JsonDeserialize(builder = EntryItemDataDTO.Builder.class)
@Value.Immutable
@ImmutableStyle
public interface EntryItemDataDTO {

    public class Builder extends EntryItemDataDTOImpl.Builder {
    }

    @Schema(required = true)
    @NotNull
    Integer getCategoryId();

    @Schema(required = true)
    @NotNull
    BigDecimal getAmount();

    @Nullable
    String getComments();

}

File: accounts/accounts-api/src/main/kotlin/com/hrrm/famoney/accounts/api/EntryItemDataDTO.kt
package com.hrrm.famoney.accounts.api

import io.swagger.v3.oas.annotations.media.Schema
import java.math.BigDecimal

@Schema(name = "EntryItemData")
interface EntryItemDataDTO {
    @get:Schema(required = true)
    val categoryId: Int

    @get:Schema(required = true)
    val amount: BigDecimal
    val comments: String?
}